File: android/capacitor/src/main/java/com/getcapacitor/plugin/WebView.java
package com.getcapacitor.plugin;

import android.app.Activity;
import android.content.SharedPreferences;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin
public class WebView extends Plugin {

    public static final String WEBVIEW_PREFS_NAME = "CapWebViewSettings";
    public static final String CAP_SERVER_PATH = "serverBasePath";

    @PluginMethod
    public void setServerBasePath(PluginCall call) {
        String path = call.getString("path");
        bridge.setServerBasePath(path);
        call.resolve();
    }

    @PluginMethod
    public void getServerBasePath(PluginCall call) {
        String path = bridge.getServerBasePath();
        JSObject ret = new JSObject();
        ret.put("path", path);
        call.resolve(ret);
    }

    @PluginMethod
    public void persistServerBasePath(PluginCall call) {
        String path = bridge.getServerBasePath();
        SharedPreferences prefs = getContext().getSharedPreferences(WEBVIEW_PREFS_NAME, Activity.MODE_PRIVATE);
        SharedPreferences.Editor editor = prefs.edit();
        editor.putString(CAP_SERVER_PATH, path);
        editor.apply();
        call.resolve();
    }
}

File: ios/Capacitor/Capacitor/Plugins/WebView.swift
import Foundation

@objc(CAPWebViewPlugin)
public class CAPWebViewPlugin: CAPPlugin {

    @objc func setServerBasePath(_ call: CAPPluginCall) {
        if let path = call.getString("path"), let viewController = bridge?.viewController as? CAPBridgeViewController {
            viewController.setServerBasePath(path: path)
            call.resolve()
        }
    }

    @objc func getServerBasePath(_ call: CAPPluginCall) {
        if let viewController = bridge?.viewController as? CAPBridgeViewController {
            let path = viewController.getServerBasePath()
            call.resolve([
                "path": path
            ])
        }
    }

    @objc func persistServerBasePath(_ call: CAPPluginCall) {
        if let viewController = bridge?.viewController as? CAPBridgeViewController {
            let path = viewController.getServerBasePath()
            let defaults = UserDefaults.standard
            defaults.set(path, forKey: "serverBasePath")
            call.resolve()
        }
    }
}
